### ISSUE TITLE
NodeJS modules

### ISSUE THREAD

**Issue by rusfearuth** [clio-lang/clio#3]
I like Clio, but I realised that only language without minimal set of modules is road to nowhere. I believe that Clio could use NodeJS modules for the first time.

And my question: does Clio have any binding for JS? If it has, will it be possible to share example?

**Comment by pouya-eghbali** [comment]
On the server-side NodeJS modules are usable and JavaScript files can be imported directly. On the browser, we have import support for JS files and Clio files, but `require` will not work in the JS files.

Right now to import and use a NodeJS module, we need to `require` them in a JS file and import this JS file from Clio (or the main file of the NodeJS module) because we do not read `package.json`. However, I will probably add this feature soon because it's nice to make it easier to use NodeJS modules.

There are a few problems with supporting JS libraries. I've used RethinkDB, jQuery, and Handlebars with Clio, but I noticed a few issues with this. In Clio, we don't have methods, instead, we have function overloading to achieve something similar. I'm not yet sure of this feature and I would like to know how the users and the community think about this so we can find the best way together.

Anyhow, when calling a function, if the function we want to call isn't available in the local scope, window or built-ins, Clio assumes it's a method and tries to call the function as a method. So for example, I'll not be able to call the `find` method of a jQuery object if I have a `find` function in my scope.

Another issue is Clio data types. I'm using Decimal.js (I'll switch to Big.js probably because it's lighter) for numbers, I want to make scientific calculations simpler and easier and for that we need a big number data type. Problem is, we cannot use these Decimal.js numbers directly in other JavaScript modules.

To be able to fully support JavaScript modules, we need to solve these problems. Other than that, JavaScript modules are usable in Clio (Clio modules can be also imported and used in JavaScript). I'll keep this issue open for discussions, and I'll open 2 more issues for the Decimal.js and method problems.
